contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```diff
@@ -401,9 +404,86 @@ PrintTupPrepareInfo(DR_printtup *myState, TupleDesc typeinfo, int numAttrs)
 	}
 }
 
+static int32
+resolve_numeric_typmod_from_append_or_mergeappend(Plan *plan, AttrNumber attno)
+{
+	ListCell	*lc;
+	uint8_t		max_precision = 0,
+				max_scale = 0,
+				precision = 0,
+				scale = 0,
+				integralDigitCount = 0;
+	int32		typmod = -1,
+				result_typmod = -1;
+	List		*planlist = NIL;
+	if (IsA(plan, Append))
+	{
+		planlist = ((Append *) plan)->appendplans;
+	}
+	else if(IsA(plan, MergeAppend))
+	{
+		planlist = ((MergeAppend *) plan)->mergeplans; 
+	}
+
+	Assert(planlist != NIL);
+	foreach(lc, ((Append *) plan)->appendplans)
```

Comment:
can you explain what do multiple plans indicate here?

Reply:
It indicates separate plan for all the subqueries that involves in UNION operation. For example,
```
select t1 from table1 
union
select t2 from table2
```
In above case, we would have two append plan one for each select .